gdk/common/recipe_file/CaseInsensitiveRecipe.py

```diff
@@ -0,0 +1,54 @@
+from requests.structures import CaseInsensitiveDict
+from gdk.common.recipe_file.Recipe import Recipe
+
+
+class CaseInsensitiveRecipe:
```

Comment:
Seems weird we are calling this a Recipe given it is not an actual Recipe. According to the OO hierarchy `CaseInsensitiveRecipe` is not a `Recipe` `isinstance(caseInsensitiveRecipe, Recipe` would return false

Maybe an Adapter? dict -> Recipe